### PR TITLE
[Tests-Only]Fixed GUI test log parser

### PR DESCRIPTION
**Pull request by Talank**
This fixes GUI test log parser to display the log from middleware steps fails as well.

### CODE REVIEW

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Comment by Talank** [comment]
It looks like the log file does not contain the error details in case of run time errors. So, I will display the entire error object in that case. If we find another appropriate method to display the error precisely then we can make changes to that. https://drone.owncloud.com/owncloud/client/8944/6/14 
Screenshot from 2021-09-15 10-27-29
In the above case, the parser fails to find the error since it can not find the keyword `detail` in the error object.

File: test/gui/TestLogParser.py [hunk 1]
```diff
@@ -56,7 +76,7 @@ def traverse_loop(data):
 
 # Remove duplicate nodes, if exists
 # This step is neccessary because sometimes the data in failing_tests_raw is redundent.
```
Comment (phil-davis) [inline test/gui/TestLogParser.py:78]:
```suggestion
# This step is necessary because sometimes the data in failing_tests_raw is redundant.
```

**Review by phil-davis — APPROVED** [review]
LGTM. If you need to rebase/change this PR again, then you can fix the typos in the existing comment.

**Pull request merged**